[Watcher] Remove `axios` dependency in tests

sabarasaba opened the pull request:
Partially addresses https://github.com/elastic/kibana/issues/127966

### Summary
In order to upgrade the axios dependency in kibana(see: #111655) we need to switch the tests from this app to use HttpSetup mock instead of axios + sinon server mock, which is technically incorrect but axios just happened to have methods signature similar to HttpSetup.

sabarasaba commented:
@elasticmachine merge upstream

elasticmachine commented:
Pinging @elastic/platform-deployment-management (Team:Deployment Management)

alisonelizabeth commented on x-pack/plugins/watcher/__jest__/client_integration/watch_create_threshold.test.tsx:
```diff
@@ -763,17 +763,13 @@ describe('<ThresholdWatchEdit /> create route', () => {
-          const requestBody = unwrapBodyResponse(latestReqToGetVisualizeData.requestBody);
+          const lastReq: any[] = httpSetup.post.mock.calls.pop() || [];
+          // Options contains two dinamically computed timestamps, so its simpler to just ignore those fields.
+          const { options, ...body } = JSON.parse(lastReq[1].body).watch;
```
It took me a minute to understand `lastReq[1]` vs. `lastReq[0]`. It might be clearer to destructure:

```
const [requestUrl, watchBody] = lastReq;
```

alisonelizabeth commented on x-pack/plugins/watcher/__jest__/client_integration/watch_create_threshold.test.tsx:
```diff
@@ -763,17 +763,13 @@ describe('<ThresholdWatchEdit /> create route', () => {
-          }
-
-          const requestBody = unwrapBodyResponse(latestReqToGetVisualizeData.requestBody);
+          const lastReq: any[] = httpSetup.post.mock.calls.pop() || [];
```
Can we avoid using `any` here?

alisonelizabeth commented on x-pack/plugins/watcher/__jest__/client_integration/watch_create_threshold.test.tsx:
```diff
@@ -23,6 +21,15 @@ const MATCH_INDICES = ['index1'];
+// against it.
+jest.mock('uuid/v4', () => {
+  return function () {
+    return '12345';
```
I noticed some of the other tests use the `WATCH_ID` constant - does it make sense to use that here (and anywhere else it is hard-coded)?

sabarasaba commented:
Thanks for the review @alisonelizabeth! I've addressed your points with https://github.com/elastic/kibana/pull/128765/commits/7c9f134736c3bec95480ab96ead2d4bf05055f11
